buf/validate/internal/lib/uri.cc

```diff
@@ -157,10 +157,7 @@ struct UriParser : ParserCommon {
   }
 
   bool consumeHost() {
-    if (str.empty()) {
-      return false;
-    }
-    if (str[0] == '[' && consumeIPLiteral()) {
+    if (!str.empty() && str[0] == '[' && consumeIPLiteral()) {
```

Comment:
Same as https://github.com/bufbuild/protovalidate-go/pull/217#discussion_r2042547983, this was returning too early. We would never enter `reg-name`, which can be empty. For example "A://".